components/nvs/include/nvs.hpp

```diff
@@ -70,6 +70,225 @@ const NvsErrCategory theNvsErrCategory{};
 } // namespace
 
 namespace espp {
+class NVSHandle : public BaseComponent {
+public:
+  /// @brief Construct a new NVSHandle object
+  /// @param[in] ns_name Namespace for NVS
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  /// @details Create an NVSHandle object for the key-value pairs in the ns_name namespace
+  explicit NVSHandle(const char *ns_name, std::error_code &ec)
+      : BaseComponent("NVSHandle", espp::Logger::Verbosity::WARN) {
+    esp_err_t err;
+    ns_name = ns_name;
```

Comment:
You should probably add a check on the ns_name here (though likely the call below will handle that for you).